sealtk/noaa/gui/Window.cpp

```diff
@@ -531,6 +562,69 @@ void WindowPrivate::setTrackModel(
   }
 }
 
+// ----------------------------------------------------------------------------
+void WindowPrivate::updateTrackSelection(QItemSelection const& selection)
+{
+  auto selectedTracks = QSet<qint64>{};
+
+  for (auto const& ri : selection.indexes())
+  {
+    auto const& mi = this->trackRepresentation.mapToSource(ri);
+    auto const& data = this->trackModel.data(mi, sc::LogicalIdentityRole);
+    selectedTracks.insert(data.value<qint64>());
+  }
+
+  this->eoWindow.player->setSelectedTrackIds(selectedTracks);
+  this->irWindow.player->setSelectedTrackIds(selectedTracks);
+}
+
+// ----------------------------------------------------------------------------
+void WindowPrivate::resetActiveTool()
+{
+  this->eoWindow.player->setActiveTool(nullptr);
+  this->irWindow.player->setActiveTool(nullptr);
+}
+
+// ----------------------------------------------------------------------------
+void WindowPrivate::createDetection(WindowData* data, QRectF const& detection)
+{
+  if (!data->trackModel)
+  {
+    data->trackModel = std::make_shared<sc::KwiverTrackModel>();
+    this->trackModel.addModel(data->trackModel.get());
+    data->player->setTrackModel(data->trackModel.get());
+  }
+
+  auto* const tracks = qobject_cast<sc::KwiverTrackModel*>(
+    data->trackModel.get());
+
+  if (tracks)
+  {
+    qint64 maxId = 0;
+    for (int row = 0; row < tracks->rowCount(); ++row)
+    {
+      auto const index = tracks->index(row, 0);
+      maxId = std::max(
+        maxId, tracks->data(index, sc::LogicalIdentityRole).toLongLong());
```

Comment:
> Actually, it would be interesting to rewrite this hunk as an algorithm...

Well... it would, if we had Ranges and could invoke `std::max_element` on a transformation. Or, better, if there was an algorithm that returned the maximum *value* (with a fallback default) instead of an iterator. Oh, well.